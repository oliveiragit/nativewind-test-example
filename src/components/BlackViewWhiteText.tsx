import { Text, View } from 'react-native';

const BlackViewWhiteText = () => {
  return (
    <View testID="black-view" className="flex-1 items-center justify-center self-stretch bg-black">
      <Text className="text-white">white text</Text>
    </View>
  );
};

export { BlackViewWhiteText };
