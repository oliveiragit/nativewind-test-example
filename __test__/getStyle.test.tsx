import { render } from '@testing-library/react-native';

import { BlackViewWhiteText } from '~/components/BlackViewWhiteText';

it('should set correct styles for black background and white text using NativeWind', () => {
  const { getByText, toJSON, getByTestId } = render(<BlackViewWhiteText />);

  const text = getByText('white text');

  const view = getByTestId('black-view');
  expect(toJSON()).toMatchSnapshot();

  expect(text).toHaveProp('className');
  expect(view).toHaveStyle({ color: 'black' }); // same rgb(0,0,0)
  expect(text).toHaveStyle({ color: 'white' }); // same rgb(255,255,255)

  /* ERROR expect(element).toHaveStyle()
    - Expected
    + Received
    - color: black;
    */
});
