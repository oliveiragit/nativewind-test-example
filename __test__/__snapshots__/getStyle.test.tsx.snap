// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should set correct styles for black background and white text using NativeWind 1`] = `
<View
  className="flex-1 items-center justify-center self-stretch bg-black"
  testID="black-view"
>
  <Text
    className="text-white"
  >
    white text
  </Text>
</View>
`;
